js/data.js

```diff
@@ -6,3 +6,15 @@ var data = {
   editing: null,
   nextEntryId: 1
 };
+
+var previousJSONEntries = localStorage.getItem('javascript-local-storage');
+if (previousJSONEntries !== null) {
+  data.entries = JSON.parse(previousJSONEntries);
```

Comment:
You should be preserving the entire `data` variable, not just `data.entries`. Otherwise, you will get multiple entries with the same id 😬

Reply:
Ahh yes! I noticed that the entry ID would restart at 1 after the page refreshed and didn't understand the logic but after your suggested fixes it makes more sense!!